pkg/chunk/cached_store.go

```diff
@@ -155,11 +155,16 @@ func (s *rSlice) ReadAt(ctx context.Context, page *Page, off int) (n int, err er
 		}
 		// partial read
 		st := time.Now()
-		in, err := s.store.storage.Get(key, int64(boff), int64(len(p)))
-		if err == nil {
-			n, err = io.ReadFull(in, p)
-			_ = in.Close()
-		}
+		page.Acquire()
+		err := utils.WithTimeout(func() error {
+			defer page.Release()
```

Comment:
Why we need acquire/release, since this is a synchronized call?

Reply:
Without the acquire the function may access released page, since it's not stopped even after timeout.

Reply:
OK I see, `utils.WithTimeout` calls func asynchronously.